table.go

```diff
@@ -486,6 +486,85 @@ func (t *Table[T]) Delete(pfx netip.Prefix) {
 	}
 }
 
+// OverlapsPrefix reports whether any IP in pfx matches a route in the
+// table.
+func (t *Table[T]) OverlapsPrefix(pfx netip.Prefix) bool {
```

Comment:
qq (cuz I would love to use this), are the semantics for overlap `len(intersection(table, pfx)) > 0` or is this more like a full inner join situation?

Reply:
The first one. Hopefully that's what the docstring says? Returns true if _any_ IP in pfx would match a route in the table.

Not sure what you mean by full inner join though. What other semantic would you want to have?

Reply:
I was thinking the other possible semantic would be the prefix is either:
 - Entirely present in the table
 - The two limits of the range are present (that is, its okay for there to be holes in the route map that the given prefix covers)